integration/kubernetes/helm-chart/alluxio/templates/logserver/log-pvc.yaml

```diff
@@ -0,0 +1,39 @@
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+{{ $enabled := .Values.logserver.enabled -}}
+{{ $needPVC := eq .Values.logserver.volumeType "persistentVolumeClaim" -}}
+{{ if and .Values.logserver.enabled $needPVC -}}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ .Values.logserver.pvcName }}
+  labels:
+    app: {{ template "alluxio.name" . }}
+    chart: {{ template "alluxio.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: alluxio-logserver
+spec:
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: {{ .Values.logserver.size }}
+  storageClassName: {{ .Values.logserver.storageClass }}
+  accessModes:
+  {{ toYaml .Values.logserver.accessModes | trim | indent 4 }}
+selector:
```

Comment:
How this will work in the case of dynamic provisioning of distributed storage? 

With Ceph, I do not provision PV as it will automatically be created. And if that automatically created PV does not have these labels, then it will not be bound to the PVC, right?

Reply:
@jiacheliu3 , the above selectors, might fail in the case of Ceph PVC.

Reply:
@nirav-chotai That's a very good point. 

**Static (manual) provision**
If we remove the `matchLabels`, it's harder to manage for users who are using static provisioning (manual). They may be creating a PV and that got mounted by undesired pods.

**Dynamic provision**
I'm not sure if it is possible to request a PV with default labels to a storageClass. Is it possible in your environment? I don't find anything online so I guess not?

If by either having `matchingLabels` or not having it, we have a hard time satisfying some use cases, do you have a better option?
Adding @madanadit to the discussion too.

Reply:
@jiacheliu3 , below is an outcome of the current setup:

```
...
spec:
  volumeMode: Filesystem
  resources:
    requests:
      storage: 1Mi
  storageClassName: standard
  accessModes:
    - ReadWriteOnce
  selector:
    matchLabels:
      app: alluxio
      release: myrelease
      heritage: Helm
      role: alluxio-worker
```
I would take care of both the scenarios with the below change:

In `log-pvc.yaml`:
```
  {{- with .Values.logserver.pvc.selector }}
  selector:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

Now, for the scenario of static/manual provision, in `values.yaml`:
```
# Remote logging server
logserver:
  enabled: true
  ...
  pvc:
    selector:
      matchLabels:
        role: alluxio-logserver
        dc: mydatacenter
        region: myregion
```
The outcome would be like this:
```
...
spec:
  volumeMode: Filesystem
  resources:
    requests:
      storage: 4Gi
  storageClassName: standard
  accessModes:
      - ReadWriteOnce
  selector:
    matchLabels:
      dc: mydatacenter
      region: myregion
      role: alluxio-logserver
```

And, for the scenario of dynamic provision, in `values.yaml`:
```
# Remote logging server
logserver:
  enabled: true
  ...
  pvc:
    selector: {}
```

The outcome would be like this:
```
...
spec:
  volumeMode: Filesystem
  resources:
    requests:
      storage: 4Gi
  storageClassName: standard
  accessModes:
      - ReadWriteOnce
```

Reply:
@nirav-chotai I'm actually working on the same as your suggestion. Will update the PR once I tested it :)